src/LocalDatastore.js

```diff
@@ -259,6 +260,65 @@ const LocalDatastore = {
     }
   },
 
+  /**
+   * Updates Local Datastore from Server
+   *
+   * <pre>
+   * await Parse.LocalDatastore.updateFromServer();
+   * </pre>
+   *
+   * @static
+   */
+  async updateFromServer() {
+    if (!this.checkIfEnabled() || this.isSyncing) {
+      return;
+    }
+    const localDatastore = await this._getAllContents();
+    const keys = [];
+    for (const key in localDatastore) {
+      if (key !== DEFAULT_PIN && !key.startsWith(PIN_PREFIX)) {
+        keys.push(key);
+      }
+    }
+    if (keys.length === 0) {
+      return;
+    }
+    this.isSyncing = true;
+    const pointersHash = {};
+    for (const key of keys) {
+      const [className, objectId] = key.split('_');
+      if (!(className in pointersHash)) {
+        pointersHash[className] = new Set();
+      }
+      pointersHash[className].add(objectId);
+    }
+    const queryPromises = Object.keys(pointersHash).map(className => {
+      const objectIds = Array.from(pointersHash[className]);
+      const query = new ParseQuery(className);
+      query.limit(objectIds.length);
+      if (objectIds.length === 1) {
+        query.equalTo('objectId', objectIds[0]);
+      } else {
+        query.containedIn('objectId', objectIds);
+      }
+      return query.find();
+    });
+    try {
+      const responses = await Promise.all(queryPromises);
+      const objects = [].concat.apply([], responses);
+      const pinPromises = objects.map((object) => {
+        const objectKey = this.getKeyForObject(object);
+        return this.pinWithName(objectKey, object._toFullJSON());
+      });
+      await Promise.all(pinPromises);
+      this.isSyncing = false;
+    } catch(error) {
+      console.log('Error syncing LocalDatastore'); // eslint-disable-line
+      console.log(error); // eslint-disable-line
```

Comment:
any reason not to use parse's logger so this stuff shows up in aggregated logs?

see middlewares, MogoTransform, PromiseRouter, etc. for examples

Reply:
something like

```
const Logger = require('parse-server/lib/logger');

Logger.logger.error(`Error syncing LocalDatastore: ${error}`);
console.log('Error syncing LocalDatastore', error);
```

Reply:
Any reason to not follow the exact pattern we have already?

i.e. 

https://github.com/parse-community/parse-server/blob/master/src/PromiseRouter.js#L10
```
import log from './logger';
...
log.error('message', object);
...
```

and remove the console.log and the eslint exception comment?

Reply:
Thats the parse-server repo, there isn't a logger in this repo.

Reply:
ha.  riiiight.  yeah, lgtm. ;) . I don't use the client js sdk is the only lame excuse i have for that one.